lib/propolis-client/src/support.rs

```diff
@@ -27,6 +27,18 @@ impl Default for Chipset {
     }
 }
 
+/// Generates a 20-byte NVMe device serial number from the bytes in a string
+/// slice. If the slice is too short to populate the entire serial number, the
+/// remaining bytes are filled with `pad`.
```

Comment:
```suggestion
/// Generates a 20-byte NVMe device serial number from the bytes in a string
/// slice. If the slice is too short to populate the entire serial number, the
/// remaining bytes are filled with `pad`.
///
/// NVMe specification (particularly 1.2 and later, see section 1.7
/// `Conventions`) requires that serial numbers are left-justified and padded
/// with spaces. We allow padding with non-spaces for the time being as a
/// flagrant violation of the spec, but as a necessary one while we decide if or
/// how we will carry forward compatibility for VMs given null-padded serials.
```
mostly want to emphasize that left-justification isn't because we think it looks nice, it's to accord with the NVMe spec. could be a comment in the function or on the test cases instead?

Reply:
Fixed in d6dd7c7.